src/main/java/seedu/contax/ui/onboarding/OnboardingUtil.java

```diff
@@ -0,0 +1,141 @@
+     * @param persons the UniquePersonList to be searched
+     * @return name of the last person
+     */
+    public static String getLastestPersonName(UniquePersonList persons) {
+        ObservableList<Person> personObservableList = persons.asUnmodifiableObservableList();
+        String newPersonName = personObservableList.get(personObservableList.size() - 1).getName().toString();
+        return newPersonName;
+    }
+
+    /**
+     * Returns the last Person in the given UniquePersonList
+     * @param persons the UniquePersonList to be searched
+     * @return Person object of the last person
+     */
+    public static Person getLatestPerson(UniquePersonList persons) {
+        ObservableList<Person> personObservableList = persons.asUnmodifiableObservableList();
+        Person person = personObservableList.get(personObservableList.size() - 1);
+        return person;
+    }
+
+    /**
+     * Checks if the given command is valid. Updates the label with error message if it is not.
+     * @param command command string to be validated
+     * @param instructionLabel label to print error message on
+     */
+    public static boolean isCommandValid(String command, OnboardingInstruction instructionLabel) {
+        try {
+            parser.parseCommand(command);
+            return true;
+        } catch (ParseException e) {
+            instructionLabel.setText(INVALID_COMMAND);
+            return false;
+        }
+    }
+
+    /**
+     * Tokenizer to be used for onboaring guide. Tokenizes the given command based on the given Onboarding step.
+     * @param step OnboardingStep to tokenize command on
+     * @param command command string to be tokenized
+     * @return
+     */
+    public static ArgumentMultimap tokenize(OnboardingStep step, String command) {
+        if (step.getOperationId() == 6) {
+            return ArgumentTokenizer.tokenize(command, PREFIX_NAME, PREFIX_PHONE,
+                    PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+        }
+        return ArgumentTokenizer.tokenize(command);
+    }
+
+    /**
+     * Processes a given user command
+     * @param step
+     * @param command
+     * @param instructionLabel
+     * @param persons
+     * @return
+     */
+    public static int processCommand(OnboardingStep step, String command,
+                                     OnboardingInstruction instructionLabel, UniquePersonList persons) {
+        ArgumentMultimap map = tokenize(step, command);
+        if (step.getOperationId() == 6) {
+            if (!map.getPreamble().equals("add")) {
+                instructionLabel.setText("Please use a add command");
+                return -1;
+            }
+
+            Person p = makePerson(map);
+            if (persons.contains(p)) {
+                instructionLabel.setText(p.getName() + " already exists! Add someone else");
+                return -1;
+            }
+
+            persons.add(p);
+        }
+        return 0;
```

Comment:
There is no need to duplicate code that already exists.
```suggestion
        Command command = parser.parseCommand(command);
        
        if (step.getOperationId() == 6) {
            if (!(command instanceof AddCommand)) {
                instructionLabel.setText("Please use a add command");
                return -1;
          }

       try {
           command.execute(model);
       } catch (CommandException ex) {
           if (ex.getMessage().equals(AddCommand.MESSAGE_DUPLICATE_PERSON)) {
               instructionLabel.setText(p.getName() + " already exists! Add someone else");
           }
           return -1;
       }
        return 0;
```